ios/brave-ios/Sources/Brave/Frontend/Browser/TabWKNavigationHandler.swift

```diff
@@ -277,35 +309,35 @@ extension BrowserViewController: WKNavigationDelegate {
     }
   }
 
-  public func webView(_ webView: WKWebView, didCommit navigation: WKNavigation!) {
-    guard let tab = tabManager[webView] else { return }
+  func webView(_ webView: WKWebView, didCommit navigation: WKNavigation!) {
+    guard let tab else { return }
 
     // Set the committed url which will also set tab.url
     tab.committedURL = webView.url
 
-    self.tabDidCommitWebViewNavigation(tab)
+    tab.navigationDelegate?.tabDidCommitWebViewNavigation(tab)
   }
 
-  public func webView(_ webView: WKWebView, didFinish navigation: WKNavigation!) {
-    guard let tab = tabManager[webView] else { return }
-    self.tabDidFinishWebViewNavigation(tab)
+  func webView(_ webView: WKWebView, didFinish navigation: WKNavigation!) {
+    guard let tab else { return }
+    tab.navigationDelegate?.tabDidFinishWebViewNavigation(tab)
```

Comment:
nit:
```swift
guard let delegate = tab?.delegate else { return }
delegate?.tabDidFinishWebViewNavigation(tab)
```

Reply:
This code has already been changed and will be reflected in a future PR anyways (removing `TabWebNavigationDelegate` and moving its methods into `TabObserver`)